x-pack/plugins/cases/server/cases_analytics_index/task_runner.ts

```diff
@@ -0,0 +1,282 @@
+      alertId: string | string[];
+      index: string | string[];
+    }>({
+      type: CASE_COMMENT_SAVED_OBJECT,
+      hasReference: references,
+      perPage: 1000,
+      sortField: 'created_at',
+      sortOrder: 'asc',
+      filter: `${CASE_COMMENT_SAVED_OBJECT}.attributes.type: ${AttachmentType.alert}`,
+      fields: ['alertId', 'index'],
+    });
+
+    for await (const findResults of finder.find()) {
+      for (const alertSO of findResults.saved_objects) {
+        const theCaseId = alertSO.references.filter(({ type }) => type === CASE_SAVED_OBJECT)[0].id;
+        const alertIds = Array.isArray(alertSO.attributes.alertId)
+          ? alertSO.attributes.alertId
+          : [alertSO.attributes.alertId];
+
+        const alertIndices = Array.isArray(alertSO.attributes.index)
+          ? alertSO.attributes.index
+          : [alertSO.attributes.index];
+
+        const alerts = alertIds.map((id, index) => ({
+          _id: id,
+          _index: alertIndices[index],
+        }));
+
+        for (const alert of alerts) {
+          casesAlertMap.set(`${alert._id}:${alert._index}`, theCaseId);
+          documents.push(alert);
+        }
+      }
+    }
+
+    finder.close();
+    const durationSOAlerts = performance.now() - startSOAlerts;
+    console.log(`SO alerts data took: ${Math.round(durationSOAlerts)}ms)`);
+
+    const startMgetAlerts = performance.now();
+    const res = await esClient.mget(
+      {
+        docs: documents,
+        _source_includes: [
+          'kibana.alert.rule.name',
+          'kibana.alert.rule.tags',
+          'host.name',
+          'host.ip',
+        ],
+      },
+      { asStream: true }
+    );
+
+    let payload = '';
+    res.setEncoding('utf8');
+    for await (const chunk of res) {
+      payload += chunk;
+    }
+
+    JSON.parse(payload);
+
+    const durationMgetAlerts = performance.now() - startMgetAlerts;
+    console.log(`Mget alerts took: ${Math.round(durationMgetAlerts)}ms)`);
+  }
+
+  private async updateIndex(cases: CaseSORes) {
+    const esClient = await this.getESClient();
+    const documents = cases.saved_objects.map((theCaseSO) => ({
+      id: theCaseSO.id,
+      title: theCaseSO.attributes.title,
+    }));
+
+    const startAggs = performance.now();
+    await this.getAttachmentAggregations(cases);
+    const durationAggs = performance.now() - startAggs;
+    console.log(`Aggs took: ${Math.round(durationAggs)}ms)`);
+    printMemoryUsage();
+
+    const startAlerts = performance.now();
+    await this.getAlertData(cases);
+    const durationAlerts = performance.now() - startAlerts;
+    console.log(`Getting alerting data took: ${Math.round(durationAlerts)}ms)`);
+    printMemoryUsage();
+
+    const startBulkUpdate = performance.now();
+    const res = await esClient.helpers.bulk({
+      datasource: documents,
+      onDocument(doc) {
+        return [
+          { update: { _id: doc.id, _index: '.cases_analytics_index_poc', retry_on_conflict: 3 } },
```

Comment:
I cannot find in this POC where the `.cases_analytics_index_poc` index is created. Did you do it manually in the devtools?